### PR TITLE
fix(ivy): template compiler should render correct $localize placeholder names

### PR DESCRIPTION
**Pull request by petebacondarwin**
The `goog.getMsg()` function requires placeholder names to be camelCased.

This is not the case for `$localize`. Here placeholder names need
match what is serialized to translation files.

Specifically such placeholder names keep their casing but have all characters
that are not in `a-z`, `A-Z`, `0-9` and `_` converted to `_`.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

File: packages/compiler/src/render3/view/i18n/localize_utils.ts [hunk 1]
```diff
@@ -37,7 +36,7 @@ class MessagePiece {
 }
 class LiteralPiece extends MessagePiece {}
 class PlaceholderPiece extends MessagePiece {
-  constructor(name: string) { super(formatI18nPlaceholderName(name)); }
+  constructor(name: string) { super(formatI18nPlaceholderName(name, false)); }
```
Comment (alxhub) [inline packages/compiler/src/render3/view/i18n/localize_utils.ts]: Consider adding a comment on the `false` argument to disambiguate it.

**Review by alxhub — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]
LGTM 👍 Thanks Pete!

**Comment by AndrewKushnir** [comment]
- [VE Presubmit](http://test/OCL:267618427:BASE:267618453:1567789060807:2eeceab6)
- [Ivy Presubmit](http://test/OCL:267618427:BASE:267618568:1567789100958:602cd88b)

**Comment by AndrewKushnir** [comment]
Merge assistance: `ci/angular: size` is a pre-existing issue affecting other PRs as well.

**Pull request closed without merging**